cmd/rm-main.go

```diff
@@ -202,10 +202,14 @@ func checkRmSyntax(ctx context.Context, cliCtx *cli.Context, encKeyDB map[string
 	for _, url := range cliCtx.Args() {
 		// clean path for aliases like s3/.
 		// Note: UNC path using / works properly in go 1.9.2 even though it breaks the UNC specification.
+		urlOrig := url
 		url = filepath.ToSlash(filepath.Clean(url))
 		// namespace removal applies only for non FS. So filter out if passed url represents a directory
 		dir := isAliasURLDir(ctx, url, encKeyDB, time.Time{})
 		if !dir {
+			if string(urlOrig[len(urlOrig)-1]) == string(os.PathSeparator) {
```

Comment:
what does this check do?

Reply:
When the url is provided with suffix `/` and with no object in it, `listAndRemove` function checks and decides that this location is empty and `fix/` is `not an object`, so errors out: 
```
mc: <ERROR> Failed to remove `mm1/test1/pre/fix/`. Object does not exist
```
To prevent this, we trim the `/` character if the location is a directory and the last character is a `/`.